set.mm

```diff
@@ -619955,6 +619979,73 @@ fixed reference functional determined by this vector (corresponding to
+      isghmd eqcomd fveq2d eleq2d biimpa adantrr rlmvsca syl6eq sylan2 lmodvscl
+      frlmvscaval oveqi fvmptd3 fvex fvmpt3i oveq2d islmhmd cop cmap c0g cfsupp
+      wbr simplr simpr fsnd elmapd mpbird cfn snfi fdmfifsupp frlmelbas sylancl
+      wb mpbir2and simpllr vex fvsng eqtr2d ex wfn ffnd fnsnbt sylc opeq2 sneqd
+      biimpd eqeq2d syl5ibrcom impbid f1o2d f1oeq3d mpbid islmim sylanbrc ) DUB
+      IZCJIZKZBEDUELZUCMIENLZUUENLZBUFZBEUUEUGMIUUDAHEUUEEUHLZUUEUHLZBUUEUILZEU
+      ILZUULNLZUUFUUFOZUUIOZUUJOUULOZUUKOUUMOZUUBEUJIZUUCUUBCUPZJIZUURCUKZDEUUS
+      JFULZUMPZUUBUUEUJIZUUCDUNZPUUDDUUKUULUUBDUUKQUUCDUBUOPUUBDUULQZUUCUUBUUTU
+      VFUVADEUUSUBJFUQUMPZURUUDAHEUSLZDUSLZEUUEBUUFDNLZUUNDUTZUVHOZDVAUUDUUREVB
+      IUVCEVCVDUUBUUEVBIZUUCUUBUVDUVMUVEUUEVCVDPUUDAUUFCAVEZLZUVJBUUDUVNUUFIZKU
+      USUVJCUVNUVPUUSUVJUVNVFZUUDUUTUVPUVQUVAUUFDEUUSUVJJUVNFUVJOZUUNVGVQVHZUUD
+      CUUSIZUVPUUCUVTUUBCJVIVHZPVJZGVKUUDUVPHVEZUUFIZKZKZCUVNUWCUVHMZLZUVOCUWCL
+      ZUVIMUWGBLZUVNBLZUWCBLZUVIMUWFUUFUVIUVHDEUUSCUBJUVNUWCFUUNUUBUUCUWEVLUUTU
+      WFUVARUUDUVPUWDVMZUUDUVPUWDVRZUUDUVTUWEUWAPUVIOUVLVNUWFUWGUUFIZUWJUWHQUWF
+      UURUVPUWDUWOUUDUURUWEUVCPUWMUWNUVHUUFEUVNUWCUUNUVLVOVPUAUWGCUAVEZLZUWHUUF
+      BJCUWPUWGSBAUUFUVOVSZUAUUFUWQVSGAUAUUFUVOUWQCUVNUWPSVTWAUWPUUFICUWPTWBVDU
+      WFUWKUVOUWLUWIUVIUWFUVPUWKUVOQUWMUWFAUUFUVOBJBUWRQUWFGRUWFUVPKCUVNTWCWDUW
+      FUWDUWLUWIQZUWNAUWCUVOUWIUUFBJCUVNUWCSZGUVPCUVNTWBVDWEWFWHUUDUVNUUMIZUWDK
+      ZKZCUVNUWCUUIMZLZUVNUWIUUJMZUXDBLUVNUWLUUJMUXCUXEUVNUWIDWGLZMUXFUXCUVNUUF
+      DUUIUXGUUSCUVJJUWCEFUUNUVRUUTUXCUVARUUDUXAUVNUVJIZUWDUUDUXAUXHUUDUUMUVJUV
+      NUUDUULDNUUDDUULUVGWIWJWKWLWMUUDUXAUWDVRZUUDUVTUXBUWAPUUOUXGOWRUXGUUJUVNU
+      WIDWNWSWOUXCUDUXDCUDVEZLZUXEUUFBJBUWRUDUUFUXKVSGAUDUUFUVOUXKCUVNUXJSVTWAC
+      UXJUXDSUXCUURUXAUWDUXDUUFIUUDUURUXBUUCUUBUUTUURUUTUUCUVARUVBWPPUUDUXAUWDV
+      MUXIUVNUUIUULUUMUUFEUWCUUNUUPUUOUUQWQVPUXCCUXDTWTUXCUWLUWIUVNUUJUXCUWDUWS
+      UXIAUWCUVOUWIUUFBUWTGCUVNXAXBVDXCWFXDUUDUUFUVJBUFUUHUUDAHUUFUVJUVOCUWCXEZ
+      UPZBGUWBUUDUWCUVJIZKZUXMUUFIZUXMUVJUUSXFMIZUXMDXGLZXHXIZUXOUXQUUSUVJUXMVF
+      UXOCUWCJUVJUUBUUCUXNXJZUUDUXNXKXLUXOUVJUUSUXMJJUXODNTUUTUXOUVARXMXNUXOUUS
+      UWCUPZUXMJUXRUXOCUWCJUYAUXTUXNUWCUYAIUUDUWCUVJVIVHXLUUSXOIUXOCXPRUXODXGTX
+      QUXOUUBUUTUXPUXQUXSKXTUUBUUCUXNVLUVAUUFDEUUSUVJUBJUXMUXRFUVRUXROUUNXRXSYA
+      UUDUVPUXNKZKZUVNUXMQZUWCUVOQZUYCUYDUYEUYCUYDKZUVOCUXMLZUWCUYDUVOUYGQUYCCU
+      VNUXMSVHUYFUUCUWCJIUYGUWCQUUBUUCUYBUYDYBHYCCUWCJJYDXSYEYFUYCUYDUYEUVNCUVO
+      XEZUPZQZUYCUUCUVNUUSYGZUYJUUBUUCUYBXJUYCUUSUVJUVNUUDUVPUVQUXNUVSWMYHUUCUY
+      KUYJCUVNYIYMYJUYEUXMUYIUVNUYEUXLUYHUWCUVOCYKYLYNYOYPYQUUDUVJUUGUUFBUVJUUG
+      QUUDUVKRYRYSUUFUUGEUUEBUUNUUGOYTUUA $.
+  $}
+
+  $( TODO: It would be nice to have a function converting free modules to
+     normed modules, a la ~ rlmnlm .  The norm of an n-dimensional vector would
+     be the square root of each of its coordinates squared.
```

Comment:
I think such a function exists already, it is `toCPreHil`. Applied to the free module over the real numbers (regarded as field) results in (generalized) Eucliedean spaces, as defined in ~df-rrx